File: tests/nodejs/cases/case_214_helper_sink_dispatch_02.js
/**
 * Task case 214: helper_sink_dispatch_02 deep stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case214HelperSinkDispatch02(input) {
  const taskName = 'helper_sink_dispatch_02';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'extreme',
    stage: 'stress',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  const sink = (obj) => retainedCache.set('helper', obj);
  // ESCAPE: helper function obscures sink dispatch.
  sink(payload);
  return 'ok';
}

module.exports = { case214HelperSinkDispatch02 };

File: tests/go/cases/case_214_helper_sink_dispatch_02.go
package escape_tests

var retainedCase214 = []map[string]string{}

func Case214HelperSinkDispatch02(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "helper_sink_dispatch_02",
		"entity": "extreme",
		"stage": "stress",
		"input": raw,
		"checksum": "helper_sink_dispatch_02:" + raw,
	}
	sink := func(obj map[string]string) { retainedCase214 = append(retainedCase214, obj) }
	// ESCAPE: helper sink function hides retention edge.
	sink(payload)
	return "ok"
}
